packages/api/src/base/index.ts

```diff
@@ -58,7 +58,7 @@ export abstract class ApiBase<ApiType extends ApiTypes> extends Getters<ApiType>
   /**
    * @description Set an external signer which will be used to sign extrinsic when account passed in is not KeyringPair
    */
-  public setSigner (signer: Signer): void {
+  public setSigner (signer: Signer | undefined): void {
```

Comment:
```suggestion
  public setSigner (signer?: Signer): void {
```

Just a small suggestion for, making `setSigner()` slightly more ergonomic

Reply:
To be clear, I am not against the `?` per se, but I did think using the union to make a user pass in an explicit arg was a cleaner API given the name and intention for this call.

Reply:
Have no strong feelings on this one either way, so ok with it as-is.